design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,257 @@
+- Close open issues (~1 week)
+
+Python CLI ~13 weeks
+
+- Python ramp-up and create design for CLI (~1 week)
+- Implement logic to fill in functionality gaps (7 CLI commands, ~3 weeks)
+  - `init`, `authn` (`login`, `authenticate`), `show `, `rotate_api_key` (`host` / `user`)
+- Write associated tests for new implementation (~3 weeks)
+- Build infrastructure for testing on new platforms (Windows, macOS environments) (~3 weeks)
+- Run a few of them on different supported platforms (Windows, macOS, REHL environments) (~2 weeks)
+- Close open issues (~1 week)
+- Add documentation
+
+Go CLI - Complete rewrite (~18+ weeks)
+
+- Create design for CLI
+- CLI Ramp up
+- Redesign CLI
+- Implement new CLI commands and logic
+- Write tests for new functionality
+- Might need to redesign and build new infrastructure for environment testing (Windows, macOS, REHL environments)
+- Run a few CLI commands on new supported platforms (Windows, macOS, REHL environments)
+- Add documentation
+
+#### Takeways and recommendation
+
+Ruby requires the least amount of work in terms of what is already fuctionally supported but its UX is not intuitive. As part of its deployment, the CLI requires a specific version of Ruby installed on the machine to run the CLI.
+
+Machines may not have internet access so we will need to pre-package Ruby and other dependencies and deliver as with an installer.
+
+Go CLI would require a full rewrite due to it's limited functionality, test coverage, and architectural flaws. 
+
+Python has most of the basic actions needed to interface with the Conjur server. We will need to fill in the gaps but is architecturally more stable and well tested. The Python CLI we deliver an executable so the user will not need install additional dependencies.
+
+##### Recommendation
+
+Python CLI because of its better design and it acts as a great point of entry. In terms of a model and architectural design, it is where we want to be so it is a great first step into reforming our APIs.
+
+## Documentation
+
+[//]: # "Add notes on what should be documented in this solution. Elaborate on where this should be documented. If the change is in open-source projects, we may need to update the docs in github too. If it's in Conjur and/or DAP mention which products are affected by it"
+
+The CLI that we choose will require documentation.
+
+Ruby CLI has documentation but we will need to update according to platforms we will support
+
+Go CLI documentation can be taken from Ruby CLI because UX is the same
+
+Python CLI hasn't been clearly documented. Therefore we will need to add documentation
+
+## Open questions
+
+[//]: # "Add any question that is still open. It makes it easier for the reader to have the open questions accumulated here istead of them being acattered along the doc"
+
+Packaging:
+
+1. For Ruby CLI, if we want to package the CLI + dependencies + Ruby how would we provide an installation package for windows / macOS? For example, in REHL we can use RPM.
```

Comment:
DAP includes limitations that prevent it from running on Windows.  Why would we need to create a package other than an RPM?

Reply:
@jvanderhoof see requirements: https://app.zenhub.com/workspaces/dap-master-no-docker-5f82b33fdb531f00128c0a07/issues/cyberark/conjur/1892